index.js

```diff
@@ -269,6 +278,9 @@ Router.prototype.handle = function handle(req, res, callback) {
       return done(layerError)
     }
 
+     // trigger the "layer found" event
+    if (events && routerEvents[1] in events) self.emit(routerEvents[1], layer, req)
```

Comment:
If we swap `req` and `layer`, all three events will conveniently start with `req` as the first argument.